evm/src/cpu/kernel/constants/mod.rs

```diff
@@ -58,16 +58,19 @@ pub(crate) fn evm_constants() -> HashMap<String, U256> {
     c.insert(CALL_STACK_LIMIT.0.into(), U256::from(CALL_STACK_LIMIT.1));
 
     for segment in Segment::all() {
-        c.insert(segment.var_name().into(), (segment as u32).into());
+        c.insert(segment.var_name().into(), (segment as u64).into());
```

Comment:
Why cast to `u64` here and `usize` below? I think the discriminant representation is `isize` (unless `#[repr(SOMETHING ELSE)]` is used) so perhaps use that? Not sure though!

Reply:
Yeah right I could use `usize` for consistency with what's below. Note however that I've had weird behaviors with the discriminant representation, like the one for `Segment` is now `usize`, but I still need to cast the segment variants to `usize`, unless I am missing something with how discriminants work.

Reply:
Discriminants are a bit under-specified in Rust imo and not super nice to work with. Can we perhaps use an explicit `repr(u64)` so that we can rest in peace wrt to weird platform specifics or any changes to how they are represented internally by the compiler in the future?

Reply:
I guess I was also wondering "why wasn't u32 enough?"

Reply:
> Can we perhaps use an explicit repr(u64) so that we can rest in peace wrt to weird platform specifics or any changes to how they are represented internally by the compiler in the future?
I'm not opposed to that (though we've been so extensively using `usize` as an equivalent of `u64` in the rest of the codebase that I don't think this alone would bring peace of heart..

As for:
> I guess I was also wondering "why wasn't u32 enough?"

That's because now, each `Segment` variant is scaled by `1 << 32`, which doesn't fit in a `u32` (obviously), hence the change in type.